library/ssl_ticket.c

```diff
@@ -331,7 +338,7 @@ int mbedtls_ssl_ticket_write(void *p_ticket,
 
     key = &ctx->keys[ctx->active];
 
-    *ticket_lifetime = ctx->ticket_lifetime;
+    *ticket_lifetime = key->lifetime;
```

Comment:
`session->ticket_lifetime` MUST be update before `mbedtls_ssl_session_save()` . Otherwise, the ticket_lifetime will be different with client received.

Reply:
The idea is that we rely solely on ssl_ticket.c to check the ticket lifetime, we do not use the ticket lifetime in `ssl_tls13_server.c`. When we create a ticket we check that its lifetime is less than MBEDTLS_SSL_TLS1_3_MAX_ALLOWED_TICKET_LIFETIME (check not done by ssl_ticket.c). When we use a ticket the parse function of `ssl_ticket.c` does the check of the ticket lifetime and we do not have to do it in `ssl_tls13_server.c`.

Reply:
> When we use a ticket the parse function of ssl_ticket.c does the check of the ticket lifetime and we do not have to do it in `ssl_tls13_server.c.`

`ticket_lifetime` is not saved in ticket if we do not set `session->ticket_lifetime` before `mbedtls_ssl_session_save()` . When we use a ticket, `ticket_lifetime` is not valid.

Reply:
`ssl_ticket.c` does have `key->lifetime` the lifetime of the key with which the ticket was encrypted. This is also the lifetime of the ticket. Thus `ssl_ticket.c` have the ticket create time, its lifetime and can check it.